File: computer3/Fall Term/Solutions/Class08_Oct_30/lvl3_mimi_and_primes.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
int n; ll g = 0, ans = 0, x;
int main(){
    ios::sync_with_stdio(0); cin.tie(0);
    cin >> n;
    for(int i=1; i <=n; i++) {
        cin >> x; g = __gcd(g, x);
    }
    for(ll i=2; i*i <= g; i++) {
        while(g % i == 0){
            ans = i; g /= i;
        }
    }
    if(g != 1) { ans = g; }
    if(ans > 0) cout << ans << "\n";
    else cout << "DNE\n";
}

File: computer3/Fall Term/Solutions/Class08_Oct_30/lvl3_mimi_and_primes.py
import sys
import math
input = sys.stdin.readline

n = int(input())
a = list(map(int, input().split()))
g, ans = 0, 0
for x in a:
    g = math.gcd(g, x)
for i in range(2, int(math.sqrt(g))+1):
    while g % i == 0:
        ans, g = i, g//i
if g != 1:
    ans = g
print(ans) if ans > 0 else print('DNE')